lib/player.rb

```diff
@@ -0,0 +1,36 @@
+require_relative 'ship'
+
+class Player
+
+  def initialize(grid)
+    @grid = grid
+  end
+
+  def place_ship(x, y, ship_name, orientation)
+    @grid.mark_ship_positions(x, y, ship_from_name(ship_name), orientation)
+  end
+
+  def attack(cell_position)
+    if !@grid.ship?(cell_position)
+      :water
+    else
+      :hit
+    end
+  end
+
+  private
+
+  def ship_from_name(ship_name)
+    case ship_name
+    when "submarine"
+      Ship.new("submarine", 1)
+    when "destroyer"
+      Ship.new("destroyer", 2)
+    when "cruiser"
+      Ship.new("cruiser", 3)
+    when "aircraft-carrier"
+      Ship.new("aircraft-carrier", 4)
+    end
```

Comment:
Sure: You could extract the "notion of creating a ship based on the name" into a separate class. That alleviates this class 😀